doc/sphinx/proof-engine/tactics.rst

```diff
@@ -102,35 +101,40 @@ specified, the default selector is used.
 Bindings
 ~~~~~~~~
 
-Tactics that take a term as an argument may also accept :token:`bindings`
-to instantiate some parameters of the term by name or position.
-The general form of a term with :token:`bindings` is
-:n:`@term__tac with @bindings` where :token:`bindings` can take two different forms:
+Tactics that take a term as an argument may also accept :token:`bindings` to
+specify the values to assign unbound variables in a term.
+Bindings can be given by position or name.  Generally these appear in the form
+:n:`with @bindings`.
 
   .. insertprodn bindings bindings
 
   .. prodn::
-     bindings ::= {+ ( {| @ident | @natural } := @term ) }
-     | {+ @one_term }
+     bindings ::= {+ @one_term }
+     | {+ ( {| @ident | @natural } := @term ) }
 
-+ In the first form, if an :token:`ident` is specified, it must be bound in the
-  type of :n:`@term` and provides the tactic with an instance for the
-  parameter of this name. If a :token:`natural` is specified, it refers to
-  the ``n``-th non-dependent premise of :n:`@term__tac`.
+.. todo PR: "must not be specified" below is correct?
 
-  .. exn:: No such binder.
-     :undocumented:
+The first form binds free variables in the left-to-right order of their first
+appearance in the relevant term.  For some tactics, all free variables
+must be provided, while other tactics automatically generate some or all
+of the bindings from the conclusion or a hypothesis.  In the latter case,
+the automatically bound free variables must not be specified.  Those tactics
```

Comment:
"Those tactics" does not sound entirely unambiguous. What about "The former"?

Reply:
Reworded to this:

```
The first form binds free variables in the left-to-right order of their first
appearance in the relevant term.  For some tactics, bindings for all free variables
must be provided, such as for :tacn:`induction`, :tacn:`destruct`, :tacn:`elim`
and :tacn:`case`.  Other tactics automatically generate some or all
of the bindings from the conclusion or a hypothesis, such as :tacn:`apply` and
:tacn:`constructor` and its variants.  In this case, only instances
for the dependent products that are not bound in the conclusion of the relevant term
are required.
```